hazelcast/src/main/java/com/hazelcast/query/impl/InternalIndex.java

```diff
@@ -75,4 +91,26 @@
      */
     PerIndexStats getPerIndexStats();
 
+    /**
+     * Get monotonically increasing stamp that confirms that the index contains
+     * only expected partitions, and that there are no concurrent partition updates, and
+     * there are no active partition updates (see {@link #beginPartitionUpdate()}).
+     * <p>
+     * Received stamp is used to verify that the index is still valid for the given
```

Comment:
Add explanation for `null` or some other special value like `-1` if you want to avoid boxing. It was a bit odd seeing you get a boxed `Long` while you can validate a primitive `long`.

Reply:
Added a reference to the `GlobalIndexPartitionTracker.getPartitionStamp` method, where the semantical meaning of the `-1` is explained.